README.md

```diff
@@ -1,19 +1,127 @@
+1. `contrib\inventory\d42_ansible_inventory_hostfile.py` can be used to create a static inventory file for ansible. You can group hosts by tags, customers, building or service level from Device42 data.
+2. `contrib\inventory\d42_ansible_dynamic_inventory.py` can be used to dynamically by ansible to get hosts from Device42 based on certain filters.
+
+-----------------------------
 ## Assumptions
+- This script works with Device42 12.0.0 and above
+### Requirements
+- ansible 2.9+
+- python 3.6.x+
+- requests (you can install it with pip install requests or apt-get install python-requests)
+- Ansible must have an available connection to your Device42 instance in order to collect devices for inventory
+
+### Installation
+#### Ansible Tower:
+* Clone the repo on the host of your Anisble server
+```bash
+git clone git@github.com:device42/ansible_device42.git
+```
+* Navigate to the cloned directory
+```bash
+cd path/to/directory
+```
+* Build the Ansible collection
+```bash
+ansible-galaxy collection build
+```
+* Install the Ansible collection 
+```bash
+ansible-galaxy collection install device42-d42-2.0.0.tar.gz -p ../.ansible/collections
+```
+
+* Set the following environmental variables (optional)
+```
+D42_USER = 'device42 user'
+D42_PWD = 'device42 password'
+D42_URL = 'https:// device42 server IP address'
+D42_SKIP_SSL_CHECK = False
+```
+
+For more information on installing collections please follow documentation here https://docs.ansible.com/ansible/latest/user_guide/collections_using.html
+
 -----------------------------
-    * This script works with Device42 12.0.0 and above
+## Inventory Plugin
+
+### Configuration
+Define an inventory file (`*.d42.yml`) within .ansible/collections/ansible_collections/device42/d42/plugins/inventory or wherever your
+Ansible collections directory is located.
+
+View documentation using `ansible-doc -t inventory device42.d42.d42`
+
+Environmental variables not defined:
+```
+plugin: device42.d42.d42
+url: https://10.10.10.10
+username: admin
+password: password
+ssl_check: False
+keyed_groups:
+    - key: d42_service_level
+      prefix: ''
+      separator: ''
+```
+
+Environmental variables defined:
+```
+plugin: device42.d42.d42
+keyed_groups:
+    - key: d42_service_level
+      prefix: ''
+      separator: ''
+```
+See [Ansible documentation](https://docs.ansible.com/ansible/latest/plugins/inventory/constructed.html) for more constructed examples.
+
+### How to run
+from the directory of your newly created file run the following command.
+
+```bash
+ansible-inventory -i *.d42.yml --graph
+```
 
-### Requirements
 -----------------------------
-    * python 2.7.x
-    * requests (you can install it with pip install requests or apt-get install python-requests)
+## Lookup Plugin
+
+### Configuration
+
+If using environmental variables skip this step, if not create a d42.py or d42_prompt.py in ansible/lib/ansible/plugins/lookup/
```

Comment:
These _should_ work with the Custom Credential configuration above or just by setting the env vars in your shell. I don't think modification of a .py file is needed.